qcodes/instrument_drivers/oxford/IPS120.py

```diff
@@ -658,41 +687,23 @@ def _get_switch_heater(self):
         Get the switch heater status.
 
         Returns:
-            result(str) : "Off magnet at zero"
-                          "On (switch open)"
-                          "Off magnet at field (switch closed)"
-                          "Heater fault (heater is on but current is low)"
-                          "No switch fitted"
+            result(str): See _GET_STATUS_SWITCH_HEATER.
         """
-        log.info('Get switch heater status')
+        self.log.info('Get switch heater status')
         result = self._execute('X')
-        status = {
-            0: "Off magnet at zero (switch closed)",
-            1: "On (switch open)",
-            2: "Off magnet at field (switch closed)",
-            5: "Heater fault (heater is on but current is low)",
-            8: "No switch fitted"}
-        return status[int(result[8])]
+        return self._GET_STATUS_SWITCH_HEATER[int(result[8])]
 
     def _set_switch_heater(self, mode):
         """
         Set the switch heater Off or On. Note: After issuing a command it is necessary to wait
         several seconds for the switch to respond.
         Args:
             mode (int) :
-            0 : "Off"
-            1 : "On, if PSU = Magnet"
-            2 : "On, No checks" (not available)
-        """
-        status = {
-            0: "Off",
-            1: "On, if PSU = Magnet"
-        }
-        if status.__contains__(mode):
-            log.info('Setting switch heater to %s' %
-                status.get(
-                    mode,
-                    "Unknown"))
+            0 : Off
+            1 : On
```

Comment:
do i understand correctly that `switch_heater` parameter has two roles - on get it returns one of the 5 statuses (as mentioned in `_GET_STATUS_SWITCH_HEATER `), but it is possible to set the switch only to `on` or `off`, meaning to `0` or `1` and not to `2`, `5`, or `8` -- is this correct? If so, this is quite dirty, but since users have been using it this way and we know of no complaints, let's keep it like this.

Reply:
You are correct, it's not pretty.